### PR TITLE
WithVolume method added to pod pkg

### PR DESCRIPTION
**Pull request by yprokule**
Method that allows to configure _ConfigMap_ volume on the pod

### CODE REVIEW

**Comment by yprokule** [comment]
Tested with next example:
```bash
    var myVolMode int32
    myVolMode = 511

    myPod = myPod.WithConfigMapVolume("configs", "yp-netconfig", &myVolMode)
```

File: pkg/pod/container.go [hunk 1]
```diff
@@ -235,6 +235,25 @@ func (builder *ContainerBuilder) WithEnvVar(name, value string) *ContainerBuilde
 	return builder
 }
 
+// WithVolumeMounts configures pod volumes to mount inside the container.
+func (builder *ContainerBuilder) WithVolumeMounts(volMounts []v1.VolumeMount) *ContainerBuilder {
```
Comment (trewest) [inline pkg/pod/container.go:239]: Can you change this to ` WithVolumeMount` and just accept a single v1.VolumeMount? Multiple mounts can be chained together.

File: pkg/pod/pod.go [hunk 2]
```diff
@@ -599,6 +599,49 @@ func (builder *Builder) WithPrivilegedFlag() *Builder {
 	return builder
 }
 
+// WithVolume attaches given volume to all pod's containers.
+func (builder *Builder) WithConfigMapVolume(volumeName, cmName string, defMode *int32) *Builder {
```
Comment (trewest) [inline pkg/pod/pod.go:603]: Since Volumes support a large amount of sources, can you just make this generic? Like `WithVolume` that accepts a v1.Volume?

Comment (trewest) [reply pkg/pod/pod.go:603]: I think it should be pretty similar to the WithVolumeMount method

**Review by trewest — CHANGES_REQUESTED** [review]
Couple of comments

File: pkg/pod/container.go (hunk 1 above)
Comment (yprokule) [reply pkg/pod/container.go]: resolved in #182

File: pkg/pod/pod.go (hunk 2 above)
Comment (cdvultur) [reply pkg/pod/pod.go:603]: Indeed, from what I know `WithLocalVolume` does use configMap.

What is does not do, is setting the `defaultMode`. 
If that is what you want, changing the signature of this function could break existing usage of that function.

Comment (yprokule) [reply pkg/pod/pod.go:603]: No `WithConfigMapVolume` should work only with _ConfigMap_ volume. `WithLocalVolume` does extra things e.g. adding _VolumeMount_ to all the containers or doesn't allow configuration of the _VolumeMount_'s name/path.

Based on @trewest comment I'll check creating of a generic method that will add _Volume_ (despite type) to the _Pod_

Comment (yprokule) [reply pkg/pod/pod.go]: @trewest updated

**Comment by yprokule** [comment]
tested with next snippet
```go
    myVolume := v1.Volume{
        Name: "configs",
        VolumeSource: v1.VolumeSource{
            ConfigMap: &v1.ConfigMapVolumeSource{
                DefaultMode: &myVolMode,
                LocalObjectReference: v1.LocalObjectReference{
                        Name: "yp-netconfig",
                },
            },
        },
    }

    myPod = myPod.WithVolume(myVolume)
```

File: pkg/pod/pod.go [hunk 3]
```diff
@@ -599,6 +599,29 @@ func (builder *Builder) WithPrivilegedFlag() *Builder {
...
+		glog.V(100).Infof("The volume's Name cannot be empty")
+
+		builder.errorMsg = "The volume's Name cannot be empty"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	glog.V(100).Infof("Configuring volume %q for pod", volume.Name)
```
Comment (trewest) [inline pkg/pod/pod.go:618]:
```suggestion
	glog.V(100).Infof("Adding volume %s to pod %s in namespace %s", volume.Name, builder.Definition.Name, builder.Definition.Namespace)
```

File: pkg/pod/pod.go [hunk 4]
```diff
@@ -599,6 +599,29 @@ func (builder *Builder) WithPrivilegedFlag() *Builder {
 	return builder
 }
 
+// WithVolume attaches given volume to a pod containers.
```
Comment (trewest) [inline pkg/pod/pod.go:602]:
```suggestion
// WithVolume attaches given volume to a pod.
```

**Review by trewest — CHANGES_REQUESTED** [review]
Two small nits, otherwise looks good

File: pkg/pod/pod.go (hunk 4 above)
Comment (yprokule) [reply pkg/pod/pod.go:602]: thanks, updated

File: pkg/pod/pod.go (hunk 3 above)
Comment (yprokule) [reply pkg/pod/pod.go]: thanks, updated

**Review by trewest — APPROVED** [review]
/lgtm

**Review by cdvultur — APPROVED** [review]
/lgtm

**Pull request merged**